solutions/find_max_in_list.py

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on 2024-12-27
+
+@author: Yuri Spizhovyi
+"""
+
+
+def find_max_in_list(numbers: list[int]) -> int | None:
+    """
+    Find and return the maximum number in a given list.
+
+    Args:
+        numbers (list[int]): A list of integers.
+
+    Returns:
+        int: The maximum number in the list.
+        None: If the list is empty.
+
+    Raises:
+        TypeError: If the input is not a list or if the list contains non-integer elements.
+
+    Tests:
+
+    >>> print(find_max_in_list([]))
+    None
+
+    >>> print(find_max_in_list([0, 0]))
+    0
+
+    >>> print(find_max_in_list([1, 2, 3]))
+    3
+
+    >>> find_max_in_list("not a list")
+    Traceback (most recent call last):
+    TypeError: Input must be a list of integers
+
+    >>> find_max_in_list([1, "2", 3])
+    Traceback (most recent call last):
+    TypeError: All elements in the list must be integers
+    """
+    if not isinstance(numbers, list):
```

Comment:
More line comments can be helpful to describe why the code is there and how it relates to other lines of the code. I think you have only one.

Reply:
Thank you for your feedback! If I understood your question correctly, the lines:
```
Traceback (most recent call last):  
TypeError: All elements in the list must be integers
```
are not comments; they are part of the doctests. These lines simulate the expected output of the code when a TypeError is raised, as required for proper test coverage. Let me know if I misunderstood or if you have additional suggestions!